src/probnum/quad/bq_methods/belief_updates/_belief_update.py

```diff
@@ -0,0 +1,114 @@
+"""Belief updates for Bayesian quadrature."""
+
+import abc
+from typing import Optional
+
+import numpy as np
+from scipy.linalg import cho_factor, cho_solve
+
+from probnum.quad.bq_methods.bq_state import BQState
+from probnum.randvars import Normal
+
+
+class BQBeliefUpdate(abc.ABC):
+    """Abstract class for the inference scheme."""
+
+    def __call__(self, *args, **kwargs):
+        raise NotImplementedError
+
+
+class BQStandardBeliefUpdate(BQBeliefUpdate):
+    """Updates integral belief and state using standard Bayesian quadrature based on
+    standard Gaussian process inference."""
+
+    def __call__(
+        self,
+        bq_state: BQState,
+        new_nodes: np.ndarray = None,
+        new_fun_evals: np.ndarray = None,
+    ):
+        """Updates integral belief and BQ state according to the new data given.
+
+        Parameters
+        ----------
+        bq_state :
+            Current state of the Bayesian quadrature loop.
+        new_nodes :
+            New nodes that have been added.
+        new_fun_evals :
+            Function evaluations at the given node.
+
+        Returns
+        -------
+        updated_belief :
+            Gaussian integral belief after conditioning on the new nodes and evaluations.
+        updated_state :
+            Updated version of ``bq_state`` that contains all updated quantities.
+        """
+
+        # Update nodes and function evaluations
+        old_nodes = bq_state.nodes
+
+        nodes = np.concatenate((bq_state.nodes, new_nodes), axis=0)
+        fun_evals = np.append(bq_state.fun_evals, new_fun_evals)
+
+        # kernel quantities
+        if old_nodes.size == 0:
+            gram = bq_state.kernel(new_nodes)
+            kernel_means = bq_state.kernel_embedding.kernel_mean(new_nodes)
+        else:
+            gram_new_new = np.atleast_2d(bq_state.kernel(new_nodes))
+            gram_old_new = np.atleast_2d(bq_state.kernel(new_nodes, old_nodes))
+            gram = np.hstack(
+                (
+                    np.vstack((bq_state.gram, gram_old_new)),
+                    np.vstack((gram_old_new.T, gram_new_new)),
+                )
+            )
+            kernel_means = np.concatenate(
+                (
+                    bq_state.kernel_means,
+                    bq_state.kernel_embedding.kernel_mean(new_nodes),
+                )
+            )
+
+        initial_error = bq_state.kernel_embedding.kernel_variance()
```

Comment:
`initial_integral_variance` to be consistent with naming? I personally don;t like error so much as it implies units of f and not f^2.